### PR TITLE
landing page code Samples v2

### PR DESCRIPTION
**Pull request by sw-yx**
## What was changed
continued iteration on code samples. followup from #35 

we abandoned signals in https://gist.github.com/sw-yx/4b8d660635ede59d8870b17b6b4e9e95 - we would like to bring it back in some way.

### CODE REVIEW

File: codesamples/sample.go [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
-	for _, interval := range intervals {
-		time.Sleep(interval * time.Day)                           // Sleep for days!
-		_ = ExecuteActivity(ctx, SendEmail, userId).Get(ctx, nil) // Activities are retried by default!
+	func RemindUserWorkflow(ctx workflow.Context, userId string, intervals []int) error {
+		// send reminder emails, e.g. after 1, 7, and 30 days
+		for _, interval := range intervals {
+			time.Sleep(interval * time.Day)                           // Sleep for days!
+			_ = ExecuteActivity(ctx, SendEmail, userId).Get(ctx, nil) // Activities are retried by default!
+		}
+		// easily modifiable to cancel if the user performs an action
```
Comment (Sushisource) [inline codesamples/sample.go:7]:
```suggestion
		// easily cancellable if the user acts
```

Derek's suggestion is good too and the "to cancel" wording is a bit awkward

File: codesamples/sample.go [hunk 2]
```diff
@@ -1,8 +1,8 @@
-func RemindUserWorkflow(ctx workflow.Context, userId string, intervals []int) error {
-	// send reminder emails, e.g. after 1, 7, and 30 days
-	for _, interval := range intervals {
-		time.Sleep(interval * time.Day)                           // Sleep for days!
-		_ = ExecuteActivity(ctx, SendEmail, userId).Get(ctx, nil) // Activities are retried by default!
+	func RemindUserWorkflow(ctx workflow.Context, userId string, intervals []int) error {
+		// Send reminder emails, e.g. after 1, 7, and 30 days
+		for _, interval := range intervals {
```
Comment (bergundy) [inline codesamples/sample.go:3]:
```suggestion
		for interval := range intervals {
```

File: codesamples/sample.go [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
+			workflow.Sleep(interval * time.Day)                                // Sleep for days!
```
Comment (bergundy) [inline codesamples/sample.go:4]: From our samples:
```suggestion
			_ = workflow.Sleep(interval * time.Day)                              // Sleep for days!
```

Comment (bergundy) [reply codesamples/sample.go:4]: https://github.com/temporalio/samples-go/blob/bca453dee53458b5a54d5a5df0e67981aa66b283/metrics/workflow.go#L21

File: codesamples/sample.java [hunk 4]
```diff
@@ -1,10 +1,10 @@
-public static class RemindUserWorkflowImpl implements RemindUserWorkflow {
-    public void EmailUser(string userId, int[] intervals) {
-      // send reminder emails, e.g. after 1, 7, and 30 days
-      for (int interval: intervals) {   
-        Workflow.sleep(Duration.ofDays(interval)); // Sleep for days!
-        activities.sendEmail(interval, userId)     // Activities retried by default!
+  public static class RemindUserWorkflowImpl implements RemindUserWorkflow {
+      public void EmailUser(String userId, int[] intervals) {
+        // Send reminder emails, e.g. after 1, 7, and 30 days
+        for (int interval: intervals) {
```
Comment (bergundy) [inline codesamples/sample.java:4]:
```suggestion
        for (int interval : intervals) {   
```

File: codesamples/sample.js [hunk 5]
```diff
@@ -1,8 +1,8 @@
 async function remindUserWorkflow(userId, intervals) {
```
Comment (bergundy) [inline codesamples/sample.js:1]: In JS the workflow main function is always called `main`.

```suggestion
// workflows/remind-user.js
async function main(userId, intervals) {
```

**Review by bergundy — COMMENTED** [review]
Added some suggestions.

Thinking about the last comment a bit more, we might want to change it from:
```
Easily modifiable to cancel if the user unsubscribes
```
To:
```
Workflow could be cancelled in case the user unsubscribes
```
Since cancellation is already built-in, there's nothing to modify.

**Pull request merged**